pkg/ingester/client/pool.go

```diff
@@ -82,33 +131,54 @@ func (pool *IngesterPool) RemoveClientFor(addr string) {
 }
 
 // RegisteredAddresses returns all the addresses that a client is cached for
-func (pool *IngesterPool) RegisteredAddresses() []string {
+func (p *Pool) RegisteredAddresses() []string {
 	result := []string{}
-	pool.RLock()
-	defer pool.RUnlock()
-	for addr := range pool.clients {
+	p.RLock()
+	defer p.RUnlock()
+	for addr := range p.clients {
 		result = append(result, addr)
 	}
 	return result
 }
 
 // Count returns how many clients are in the cache
-func (pool *IngesterPool) Count() int {
-	pool.RLock()
-	defer pool.RUnlock()
-	return len(pool.clients)
+func (p *Pool) Count() int {
+	p.RLock()
+	defer p.RUnlock()
+	return len(p.clients)
+}
+
+func (p *Pool) removeStaleIngesterClients() {
```

Comment:
You seem to be mostly removing the word 'ingester' from this file, so why re-introduce it here?